File: LeetCode Solutions/March 2022 LeetCoding Challenges/Score of Parentheses/Score of Parentheses.cpp
class Solution {
public:
    int scoreOfParentheses(string s) {
        int power = 0, score = 0;
        for (int i=1; i<s.length(); i++){
            if (s[i] == '('){
                power++;
            }
            else if (s[i-1] == '('){
                score += 1 << power--;
            } else{
                power--;
            }
        }
        return score;
    }
};

File: LeetCode Solutions/LeetCoding Challenge/February 2020 LeetCoding Challenge/Score of Parentheses.py
class Solution:
    def scoreOfParentheses(self, S: str) -> int:
        power, score = 0, 0
        for i in range(1, len(S)):
            if(S[i] == "("):
                power += 1
            elif(S[i-1] == "("):
                score += 1 << power
                power -= 1
            else:
                power -= 1
        return score